combat-trainer.lic

```diff
@@ -355,11 +355,25 @@ class LootProcess
   end
 
   def should_perform_ritual(prepare_consume)
-    !prepare_consume &&
-      ((@necro_train_first_aid && DRSkill.getxp('First Aid') < 32) || DRSkill.getxp('Thanatology') < 32)
+   !prepare_consume &&
+    ((@ritual_type == "dissect" && DRSkill.getxp('First Aid') < 32) ||
+    (@ritual_type == "harvest" && DRSkill.getxp('Skinning') < 32) ||
+    DRSkill.getxp('Thanatology') < 32)
+  end
+
+  def check_ritual_xp
+    DRSkill.getxp('First Aid') > 31 && DRSkill.getxp('Skinning') > 31
+  end
+
+  def cycle_rituals
```

Comment:
I think a lot of this logic can be merged with `should_perform_ritual`. The combined method would be something like `determine_ritual` and return whatever is appropriate based on cycle/no-cycle and experience. `nil` (or empty string) could represent 'do not do a ritual' (i.e. the `false` value right now).